source/elements/oneMKL/source/domains/spblas/matrix_view.rst

```diff
@@ -0,0 +1,134 @@
+.. SPDX-FileCopyrightText: 2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_matrix_view_and_descr:
+
+Matrix view
+===========
+
+.. _onemkl_sparse_matrix_descriptor:
+
+matrix_descr
+------------
+
+.. rubric:: Definition
+
+.. code:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       enum class matrix_descr {
+           general,
+           symmetric,
+           hermitian,
+           triangular,
+           diagonal,
+       };
+
+   }
+
+.. container:: section
+
+   .. rubric:: Description
+
+   The matrix descriptor is an optional property that can optimize some
+   operations.
+
+   TODO: Clarify are triangular operations going to require a triangular
+   matrix_descr or should it be ignored?
```

Comment:
@gajanan-choudhary, how were you planning to handle the error checking with `matrix_view`? I think it would be easier to ignore the `type_view` field for some operations, like `trsv`.

Reply:
Alternatively it could make more sense for ~~`trsv`~~ `spsv` to have an `uplo` parameter instead of a `matrix_view`? It's not consistent with other operations but it avoids some unused parameters.

Reply:
Yes, Triangular operations such as SpSV require a triangular "view" (`matrix_descr`), and they must specify which triangular (lower/upper) as well.

Reply:
I have removed the TODO and added the restriction in `spsv` documentation in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/90c4cec984f71f62b1f60d19080cabd3ae643408.
I have specified that a `diagonal` matrix is allowed as well, I expect this could be useful?